pom.xml

```diff
@@ -62,6 +62,13 @@
           </dependency>
         </dependencies>
       </plugin>
+      <plugin>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <configuration>
+          <forkCount>0.8C</forkCount>
```

Comment:
on a high CPU count machine that are not servers (e.g. laptops) this is an large amount of memory :-)  (9GB on a 16 core machine)

generally it is better to configure this where you know the environment (like CI) by passing a flag to `-DforkCount`

also and a big issue is specifying it here in the configuration means it takes priority over any user specified argument so I for example have no option but to have 12 surefire JVMs from the command line and anyone that is running tests in a different environment that had tweaked surefire by knowing the environment is similarly impacted.

Reply:
I don't think that I have that level of control with the `buildPlugin` Jenkinsfile that is used in this repository.

I'd like to continue using `buildPlugin` and I'd like to be able to run tests in parallel on multi-core computers.  Is it possible for me to have both those conditions and still satisfy your concerns or am I "stuck" that I can only keep the Jenkinsfile and satisfy your concerns by running tests serially?

Reply:
you can use a profile that is enabled on an environment variable being present (and you can use one of the ones that Jenkins exists, e.g. `BUILD_URL`).

in the profile set the property `forkCount` , then remove this configuration from the pom.

Alternatively - remove this completely and raise an issue in the INFRA project to set it based on the underlying hardware, and allow the configuration to be exposed where needed.

```xml

  <profiles>
    <profile>
      <!-- use 0.8C forks when running a build in Jenkins -->
      <id>jenkins</id>

      <activation>
        <property>
          <name>env.BUILD_URL</name>
        </property>
      </activation>
      <propterties>
        <forkCount>0.8C</forkCount>
      </propterties>
    </profile>
  </profiles>

```